sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -798,4 +798,27 @@ public void testJoinOnVirtualTableWithSingleRowSubselect() throws Exception {
                 "        unnest([2]) tt2");
         assertThat(printedTable(response.rows()), is("1| 1| 2\n"));
     }
+
+    @Test
+    public void testFourTableJoin() {
```

Comment:
Please make a test in ManyTableConsumerTest instead of a full-blown integ test.

Reply:
I'm a bit hesitant to do that. Here's why: 

1)  This test is for a soon unmaintained branch and won't be part of future versions of Crate. Therefore, the additional overhead of a full integration test is not an issue here. 
2) The integration test is not that expensive since it is part of a suite where the cluster is only broad up once.
3) The test is valid. It checks the result of the query and gives additional confidence since four table joins don't seem to be fully integration tested yet.

Reply:
@mxm please rename the test method that shows the actual scenario ("testJoinWithConditionsReferringToTheFirstTable" is quite big but something descriptive)